pkg/nmi/server/server.go

```diff
@@ -125,6 +132,49 @@ func (fn appHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	latency := time.Since(start)
 	logger.Infof("Status (%d) took %d ns", rw.statusCode, latency.Nanoseconds())
 }
+func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.Request) {
+	hostIP := parseRemoteAddr(r.RemoteAddr)
+	 if hostIP != "127.0.0.1" {
+		msg := "request remote address is not from a host"
+		logger.Error(msg)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	 }
+
+	 podns, podname := parseRequstHeader(r)
+	 if podns == "" || podname == "" {
+		 logger.Errorf("Error getting podname and podns from request")
+		 http.Error(w, "Error getting podname and podns from request", http.StatusInternalServerError)
```

Comment:
this is not internalserver error but a error on the client side of bad request, adding info in the error to how to send the pod name/namespace will make this more useful

Reply:
Good idea... should i use `StatusUnprocessableEntity` 422 instead? 

Just FYI, in `msiHandler` we currently return `StatusInternalServerError` when podns and podname cannot be found. should we also return a diff status to be consistent? 
```
podns, podname, err := s.KubeClient.GetPodName(podIP)
	if err != nil {
		logger.Errorf("missing podname for podip:%s, %+v", podIP, err)
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
```

Reply:
this one is ok since the request came from the pod but server was not able to find the pod